### PR TITLE
Clean up location defaulting

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

For consistency, default `state_code` and `iecc_zone` in `hpxml_defaults.rb` instead of `BuildResidentialHPXML` (like `site_type`). Update tasks.rb, docs, changelog, EPvalidator.xml, etc.

## Checklist

Not all may apply:

- [x] EPvalidator.xml has been updated
- [x] Tests (and test files) have been updated
- [x] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

File: docs/source/workflow_inputs.rst [hunk 1]
```diff
@@ -192,6 +192,16 @@ A couple schedule CSV file examples are provided in the ``HPXMLtoOpenStudio/reso
...
 
+Generator inputs are entered in ``/HPXML/Building/Site/Address``.
+
+  =================================  ========  =====  ===========  ========  ========  ===============
+  Element                            Type      Units  Constraints  Required  Default   Description
+  =================================  ========  =====  ===========  ========  ========  ===============
+  ``StateCode``                      string                        No        EPW [#]_  Site state code
+  =================================  ========  =====  ===========  ========  ========  ===============
+
+  .. [#] State code will be defined according to the EPW weather file header.
```
Comment (shorowit) [inline docs/source/workflow_inputs.rst:203]: Please incorporate this into the existing `HPXML Site` section of the documentation.

File: tasks.rb [hunk 2]
```diff
@@ -493,6 +493,8 @@ def set_measure_argument_values(hpxml_file, args)
 
   if ['base.xml'].include? hpxml_file
     args['simulation_control_timestep'] = 60
+    args['site_iecc_zone'] = '5B'
+    args['site_state_code'] = 'CO'
```
Comment (shorowit) [inline tasks.rb:497]: Does this actually change anything? And why don't I see updated HPXML files with these values?

**Review by shorowit — COMMENTED** [review]
Please review the PR checklist. This needs, at least, updates to EPvalidator.xml and new tests in test_defaults.rb.

File: docs/source/workflow_inputs.rst (hunk 1 above)
Comment (joseph-robertson) [reply docs/source/workflow_inputs.rst:203]: But keep `state_code` and `zip_code` in the Header class, right? (There is `/Building/Site` and there is `Building/BuildingDetails/BuildingSummary/Site`.)

File: tasks.rb (hunk 2 above)
Comment (joseph-robertson) [reply tasks.rb:497]: It doesn't change anything because the `BuildResidentialHPXML` measure was already filling these in. Calling these arguments out in `tasks.rb` keeps them populated in the sample files.

Comment (shorowit) [reply tasks.rb:497]: Ah, that makes sense. Thanks.

File: docs/source/workflow_inputs.rst (hunk 1 above)
Comment (shorowit) [reply docs/source/workflow_inputs.rst:203]: Oh, good point, I didn't realize that. In that case, I suggest creating a new "HPXML Building Site" section above (and at the same level as) "HPXML Building Summary". You could also link to that section from here.

Comment (shorowit) [reply docs/source/workflow_inputs.rst:203]: (Seems like zip code needs to be documented as an optional input too.)

And I'm fine to keep them both in the HPXML Header class for simplicity.

File: Changelog.md [hunk 3]
```diff
@@ -9,6 +10,7 @@ __New Features__
 - Relaxes requirement for `ConditionedFloorAreaServed` for air distribution systems; now only needed if duct surface areas not provided.
 
 __Bugfixes__
+- Perform defaulting of state code and IECC climate zone in `hpxml_defaults.rb` instead of the BuildResidentialHPXML measure.
```
Comment (shorowit) [inline Changelog.md:13]: Is this really a bugfix? Doesn't seem like it. I recommend removing it.

File: HPXMLtoOpenStudio/resources/hpxml_defaults.rb [hunk 4]
```diff
@@ -188,6 +189,13 @@ def self.apply_header(hpxml, epw_file)
       hpxml.header.use_max_load_for_heat_pumps = true
       hpxml.header.use_max_load_for_heat_pumps_isdefaulted = true
     end
+
+    if (not epw_file.nil?) && hpxml.header.state_code.nil?
+      if Constants.StateCodes.include? epw_file.stateProvinceRegion
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/hpxml_defaults.rb:194]: Why restrict the defaults here? This doesn't seem necessary.

File: HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml [hunk 5]
```diff
@@ -93,6 +94,15 @@
...
... or text()="AR" or text()="AZ" or text()="CA" or text()="CO" or text()="CT" or text()="DC" or text()="DE" or text()="FL" or text()="GA" or text()="HI" or text()="IA" or text()="ID" or text()="IL" or text()="IN" or text()="KS" or text()="KY" or text()="LA" or text()="MA" or text()="MD" or text()="ME" or text()="MI" or text()="MN" or text()="MO" or text()="MS" or text()="MT" or text()="NC" or text()="ND" or text()="NE" or text()="NH" or text()="NJ" or text()="NM" or text()="NV" or text()="NY" or text()="OH" or text()="OK" or text()="OR" or text()="PA" or text()="RI" or text()="SC" or text()="SD" or text()="TN" or text()="TX" or text()="UT" or text()="VA" or text()="VT" or text()="WA" or text()="WI" or text()="WV" or text()="WY"] or not(h:StateCode)'>Expected StateCode to be 'AK' or 'AL' or 'AR' or 'AZ' or 'CA' or 'CO' or 'CT' or 'DC' or 'DE' or 'FL' or 'GA' or 'HI' or 'IA' or 'ID' or 'IL' or 'IN' or 'KS' or 'KY' or 'LA' or 'MA' or 'MD' or 'ME' or 'MI' or 'MN' or 'MO' or 'MS' or 'MI' or 'NC' or 'ND' or 'NE' or 'NH' or 'NJ' or 'NM' or 'NV' or 'NY' or 'OH' or 'OK' or 'OR' or 'PA' or 'RI' or 'SC' or 'SD' or 'TN' or 'TX' or 'UT' or 'VA' or 'VT' or 'WA' or 'WI' or 'WV' or 'WY'</sch:assert>
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml:101]: I would prefer that we not restrict these values, since users from another country (Canada, etc.) might want to provide a value that isn't a US state.

File: workflow/sample_files/base-location-capetown-zaf.xml [hunk 6]
```diff
@@ -45,10 +45,6 @@
         </BuildingConstruction>
       </BuildingSummary>
       <ClimateandRiskZones>
-        <ClimateZoneIECC>
-          <Year>2006</Year>
-          <ClimateZone>3A</ClimateZone>
-        </ClimateZoneIECC>
```
Comment (shorowit) [inline workflow/sample_files/base-location-capetown-zaf.xml:51]: Why did you remove this?

**Review by shorowit — COMMENTED** [review]
This is a really nice set of changes, thanks for the cleanup. Just a few lingering items to resolve.

File: HPXMLtoOpenStudio/resources/hpxml_defaults.rb (hunk 4 above)
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/hpxml_defaults.rb:194]: `base-location-capetown-zaf.xml` returns a `-`, for example.

File: workflow/sample_files/base-location-capetown-zaf.xml (hunk 6 above)
Comment (joseph-robertson) [reply workflow/sample_files/base-location-capetown-zaf.xml:51]: Because that's not the correct climate zone for Capetown.

File: HPXMLtoOpenStudio/resources/hpxml_defaults.rb (hunk 4 above)
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/hpxml_defaults.rb:194]: Maybe we restrict the defaults to being at least a single alpha character, or something?

Comment (shorowit) [reply HPXMLtoOpenStudio/resources/hpxml_defaults.rb:194]: I don't see why we can't just use whatever is in the epw file. What is the harm?

File: workflow/sample_files/base-location-capetown-zaf.xml (hunk 6 above)
Comment (shorowit) [reply workflow/sample_files/base-location-capetown-zaf.xml:51]: So why not change it to the correct value?

File: HPXMLtoOpenStudio/resources/hpxml_defaults.rb (hunk 4 above)
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/hpxml_defaults.rb:194]: Seems weird to default to a "-" character for a "state code"...

File: workflow/sample_files/base-location-capetown-zaf.xml (hunk 6 above)
Comment (joseph-robertson) [reply workflow/sample_files/base-location-capetown-zaf.xml:51]: Is IECC climate zone defined for Capetown?

File: HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml (hunk 5 above)
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml:101]: image

Comment (shorowit) [reply HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml:101]: "can include"

Comment (shorowit) [reply HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml:101]: Also, the EPvalidator proposal doesn't allow Guam or some of these other locations.

Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml]: I removed it.

File: HPXMLtoOpenStudio/resources/hpxml_defaults.rb (hunk 4 above)
Comment (joseph-robertson) [reply HPXMLtoOpenStudio/resources/hpxml_defaults.rb]: I put in a check that `epw_file.stateProvinceRegion` is two alpha characters.

File: workflow/sample_files/base-location-capetown-zaf.xml (hunk 6 above)
Comment (shorowit) [reply workflow/sample_files/base-location-capetown-zaf.xml:51]: This change is probably fine, I just didn't understand what prompted you to make it. We can go with it.

**Review by shorowit — APPROVED** [review]
Looks good to me! I'll merge this in when the CI comes back green.
